### PR TITLE
ZIO Test: Add ZIO Test to Microsite

### LINKED ISSUE #1550: Add microsite section about ZIO Test
Explain how to use ZIO Test, how to run it with sbt, etc.

### PR DESCRIPTION
**Pull request by adamgfraser**
Resolves #1550. Feedback much appreciated.

### CODE REVIEW

**Comment by adamgfraser** [comment]
@softinio Thanks! Ready to go from my end. @ghostdogpr @jdegoes Any comments?

**Comment by ghostdogpr** [comment]
Maybe say something about what needs to be added in build.sbt in order to use it (extra dependencies + use sbt test framework)?

File: docs/usecases/testing.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+When tests do fail, it is easy to see what went wrong because the test reporter will show you the entire assertion that failed and the specific part of the assertion that failed. To facilitate this, a variety of assertion combinators are included in the `Assertion` class.
+
+**Test Aspects**
+
+Test aspects are powerful tools for modifying behavior of individual tests or even entire suites that you have already written. Convenient syntax `@@` is provided for applying test aspects. So for example you could apply a timeout to a test by using `test @@ timeout(60.seconds)` or only run a test on javascript by using `test @@ jsOnly`. Test aspects are highly composable so you can combine multiple test aspects together or apply them only to certain tests that match a predicate you specify.
+
+**Zero Dependencies**
+
+As a library with zero third party dependencies, this project is availble on the JVM, ScalaJS, and will be available on Dotty and Scala Native in the near future. So you can write your tests once and make sure that your code works correctly across all platforms that you support.
```
Comment (ghostdogpr) [inline docs/usecases/testing.md]: Typo: available

**Comment by adamgfraser** [comment]
Good comments. I will submit a follow up.

File: docs/usecases/testing.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
```
Comment (ghostdogpr) [inline docs/usecases/testing.md]: This page also needs to be added in https://github.com/zio/zio/blob/master/website/sidebars.json

**Pull request merged**